velox/functions/sparksql/Arithmetic.h

```diff
@@ -483,4 +484,94 @@ struct RIntFunction {
     result = std::rint(input);
   }
 };
+
+template <typename TExec>
+struct CheckedAddFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+  template <typename T>
+  FOLLY_ALWAYS_INLINE Status call(T& result, const T& a, const T& b) {
+    if constexpr (std::is_integral_v<T>) {
+      T res;
+      bool overflow = __builtin_add_overflow(a, b, &res);
+      if (UNLIKELY(overflow)) {
+        if (threadSkipErrorDetails()) {
+          return Status::UserError();
+        }
+        return Status::UserError(
+            "[ARITHMETIC_OVERFLOW] overflow: {} + {}", a, b);
```

Comment:
nit: I wonder why we are adopting this style for the error message. It appears that something like `"Arithmetic overflow: {} + {}."` is used more frequently.

Reply:
Below is error message from Spark, I want to align the error with Spark's (`[ARITHMETIC_OVERFLOW]`)..
`[ARITHMETIC_OVERFLOW] integer overflow. Use 'try_add' to tolerate overflow and return NULL instead. If necessary set spark.sql.ansi.enabled to "false" to bypass this error.`

Reply:
> I want to align the error message with Spark's so I use the prefix.

For error messages, I feel it's better to adopt Velox's style rather than Spark's.

Reply:
I'm ok, updated, thanks.